sys/include/net/gnrc/ipv6.h

```diff
@@ -145,13 +145,14 @@ kernel_pid_t gnrc_ipv6_init(void);
  * This situation may happen when the packet has a source routing extension
  * header (RFC 6554), and the packet is forwarded from an interface to another.
  *
- * @param[in] netif     The receiving interface.
- * @param[in] current   A snip to process.
- * @param[in] pkt       A packet.
- * @param[in] nh        A protocol number (see @ref net_protnum) of the current snip.
+ * @param[in] netif      The receiving interface.
+ * @param[in] current    A snip to process.
+ * @param[in] pkt        A packet.
+ * @param[in] nh         A protocol number (see @ref net_protnum) of the current snip.
+ * @param[in] is_for_me  Indicates if we are the destination of the packet.
  */
 void gnrc_ipv6_demux(gnrc_netif_t *netif, gnrc_pktsnip_t *current,
-                     gnrc_pktsnip_t *pkt, uint8_t nh);
+                     gnrc_pktsnip_t *pkt, uint8_t nh, bool is_for_me);
```

Comment:
I have some refactoring work regarding extension headers laying around (not even pushed yet) that would make this additional parameter most likely unnecessary. This would however throw back this PR again... So I think it makes more sense to merge this first and then clean it up with my refactoring ^^.

Reply:
There is a lot of weird stuff happening with the extension headers at the moment that can be implemented much more straight forward, that is what the refactoring is for ;-)

Reply:
It would be great to merge this aged PR finally.
This one (or equivalent) is required to start implementing Non-Stroing MOP for RPL.

Reply:
I know :-/, but I also finally wanted to clean-up IPv6 reception. I can help you to adapt it to the refactoring!

Reply:
(Then someone else of course needs to help reviewing)

Reply:
Yeah, I'm convinced that it doesn't make sense to merge this PR now.
There's no use to have this extension header handling in, since now it would be more an obstacle then helpful.
I will wait and refactor it to the coming changes :)